client/client.go

```diff
@@ -141,12 +141,19 @@ func (c *Client) SubscribeHeaders(ctx context.Context, results chan<- Header) er
 	}
 
 	// start a go routine to parse the incoming data and feed to the results channel
-	go func(in <-chan interface{}) {
-		for elem := range in {
-			// TODO: return actual transaction content as well? not just ID and Result
-			// TODO: safer casting???
-			val := elem.(tmtypes.EventDataNewBlockHeader)
-			results <- val.Header
+	go func(in <-chan ctypes.ResultEvent) {
+	EventLoop:
+		for {
+			select {
+			case <-ctx.Done():
+				break EventLoop
+			case msg := <-in:
+				// TODO: return actual transaction content as well? not just ID and Result
+				// TODO: safer casting???
+				val := msg.Data.(tmtypes.EventDataNewBlockHeader)
+				results <- val.Header
```

Comment:
I think this should also do a `select` together with `ctx.Done()`

Reply:
Yeah, that's what I fixed it to do as well